compiler_gym/util/minimize_trajectory.py

```diff
@@ -25,10 +24,10 @@ class MinimizationError(OSError):
 
 # A hypothesis is a callback that accepts as input an enivornment in a given
 # state returns true if a particular hypothesis holds, else false.
-Hypothesis = Callable[[CompilerEnv], bool]
+Hypothesis = Callable[["CompilerEnv"], bool]  # noqa: F821
```

Comment:
Using a string literal type hint defers its evaluation. Otherwise `CompilerEnv` name would be evaluated here. This is used for forward references or, in this case, to break a circular reference.

As of py3.7 there is a better way of handling this, but CompilerGym supports py3.6: https://www.python.org/dev/peps/pep-0563